sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -199,7 +202,37 @@ class SQLAppStatusListener(
   }
 
   private def aggregateMetrics(exec: LiveExecutionData): Map[Long, String] = {
-    val metricTypes = exec.metrics.map { m => (m.accumulatorId, m.metricType) }.toMap
+    val accumIds = exec.metrics.map(_.accumulatorId).toSet
+
+    val metricAggregationMap = new mutable.HashMap[String, (Array[Long], Array[Long]) => String]()
+    val metricAggregationMethods = exec.metrics.map { m =>
+      val optClassName = CustomMetrics.parseV2CustomMetricType(m.metricType)
+      val metricAggMethod = optClassName.map { className =>
+        if (metricAggregationMap.contains(className)) {
+          metricAggregationMap(className)
+        } else {
+          // Try to initiate custom metric object
+          try {
+            val metric = Utils.loadExtensions(classOf[CustomMetric], Seq(className), conf).head
```

Comment:
Hi @viirya, the `loadExtensions` requires that class has a no-arg constructor, or a 1-arg constructor that takes a `SparkConf`, do we really need this restriction for `CustomMetric`?

Reply:
For current usage, I don't see there are other necessary args for `CustomMetric` constructor. Besides, as an interface, we cannot define constructor for `CustomMetric`. I think `CustomMetric` should be pretty simple class and I'd like to keep it as simple as possible. If there are more usages that require additional args, we can add an initialization API used to pass some parameters.  But I doubt that if it is necessary.

Reply:
I agree that we should keep it as simple as possible. I'm asking because the simple case class does not work since it doesn't have a no-arg constructor, or a 1-arg constructor.
```
case class SumMetric(name: String, description: String) extends CustomSumMetric
```
It took me a while to dig out because it just doesn't work and no WARN/ERROR logs.

Reply:
Good point. Added a warning log at #37386